c/httpfileservice.c

```diff
@@ -197,6 +197,35 @@ void deleteUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
   }
 }
 
+void createFileFromUnixDirectoryAndRespond(HttpResponse *response, char *absolutePath) {
+  if (isDir(absolutePath)) {
+      int folderNameLen = strlen(absolutePath);
+      if(folderNameLen == 0 || absolutePath == NULL) {
+         respondWithJsonError(response, "Failed to idenity the folder pointed", 400, "Bad Request");
+      }
+      int slashPos = lastIndexOf(absolutePath, folderNameLen, '/');
+      char *tarFileName = (slashPos == -1) ? absolutePath : absolutePath + slashPos + 1;
+      char finalFileName[strlen(tarFileName) + strlen(".tar")];
```

Comment:
I suggest you avoid using VLA. There is a file name limit, you can use that.

Reply:
@ifakhrutdinov  am not familiar with C what is VLA  ?

Reply:
sure